docs/book/src/topics/irsa-and-oidc-provider.md

```diff
@@ -0,0 +1,107 @@
+# Service Account Issuer Discovery and AWS IAM Roles for Service Accounts (IRSA)
+
+## Overview
+
+The Cluster API for AWS provider can optionally publish the Kubernetes service account token issuer to a public location
+and configure AWS to trust it to authenticate Kubernetes service accounts. 
+
+This enables the IAM Roles for Service Accounts (IRSA) capability on the cluster, allowing workloads running within the
+cluster to authenticate to the AWS API and/or to 3rd party APIs that are able to trust OIDC providers.
+
+This document applies only to unmanaged Kubernetes clusters created by this provider. EKS does these actions automatically for
+Managed clusters.
+
+## How to enable
+
+### 1 - Enable Experiment feature flag
+
+* The creation of S3 buckets and association of OIDC provider is not allowed unless the following experimental feature flags is enabled:
+    * `OIDCProviderSupport` (`EXP_OIDC_PROVIDER_SUPPORT`)
+
+### 2 - Set S3 Bucket configuration
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+kind: AWSCluster
+metadata:
+  name: capa-quickstart
+spec:
+  ...
+  s3Bucket:
+    controlPlaneIAMInstanceProfile: control-plane.cluster-api-provider-aws.sigs.k8s.io
+    nodesIAMInstanceProfiles:
+      - nodes.cluster-api-provider-aws.sigs.k8s.io
+    name: my-cluster-bucket
+```
+
+### 3 - Enable `spec.associateOIDCProvider`
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+kind: AWSCluster
+metadata:
+  name: capa-quickstart
+spec:
+  ...
+  associateOIDCProvider: true
+```
+
+
+### 4 - Set Service Account Issuer URL in KubeadmControlPlane Configuration
+
+The KubeadmControlPlane configuration should be updated to set the apiServer `service-account-issuer` argument with the S3 buckets URL:
```

Comment:
It might also be nice to create a new cluster flavor (i.e. template) in the **templates** directory.

Reply:
A new cluster flavor has been added.